pkg/registry/apis/provisioning/controller/finalizers.go

```diff
@@ -113,27 +117,73 @@ func (f *finalizer) processExistingItems(
 
 	// Safe deletion order
 	sortResourceListForDeletion(items)
-	count := 0
 
+	var dashboards, folderItems []*provisioning.ResourceListItem
```

Comment:
can we add some concurrency tests here for the concurrent logic? It should be possible as we mock the client.